csr/study_registry_reader.py

```diff
@@ -0,0 +1,36 @@
+import logging
+
+from csr.csr import CentralSubjectRegistry, Individual, Diagnosis, Biosource, Biomaterial
+from csr.entity_reader import EntityReader
+
+logger = logging.getLogger(__name__)
+
+
+class SubjectRegistryReader(EntityReader):
+    """Reader that reads Central Subject Registry (CSR) data from tab delimited files.
+    """
+
+    def __init__(self, input_dir: str):
+        EntityReader.__init__(self, input_dir)
+
+    def read_subject_registry(self) -> CentralSubjectRegistry:
+        try:
+            individual_dicts = self.read_entities('{}/{}'.format(self.input_dir, 'individual.tsv'), Individual.schema())
+            individuals = list(map(lambda i: Individual(**i), individual_dicts))
```

Comment:
These transformations to data class instances could also be moved to the read_entities function. If you pass the entity type as argument then that can be used as constructor (and also to request the schema).
E.g., 
```python
def read_entities(self, file_name: str, entity_type: Any) -> List[Any]:
    ...
    return list(map(lambda e: entity_type(**e), values_df.to_dict('records')))
```